controllers/openstackvmset_controller.go

```diff
@@ -675,6 +690,66 @@ func (r *OpenStackVMSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, nil
 }
 
+func (r *OpenStackVMSetReconciler) generateVMSetFencingKubeconfig(instance *ospdirectorv1beta1.OpenStackVMSet, labels map[string]string, envVars *map[string]common.EnvSetter) error {
+	// Get the kubeconfig used by the operator to acquire the cluster's API address
+	kubeconfig, err := config.GetConfig()
+
+	if err != nil {
+		return err
+	}
+
+	templateParameters := map[string]interface{}{}
+	templateParameters["Server"] = kubeconfig.Host
+	templateParameters["Namespace"] = instance.Namespace
+
+	// Find the API token for the osp-director-operator-kubevirtagent service account
+	kubevirtAgentServiceAccount, err := r.Kclient.CoreV1().ServiceAccounts(instance.Namespace).Get(context.TODO(), vmset.KubevirtFencingServiceAccount, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	tokenSecretName := ""
+
+	for _, secret := range kubevirtAgentServiceAccount.Secrets {
+		if strings.Contains(secret.Name, fmt.Sprintf("%s-token", vmset.KubevirtFencingServiceAccount)) {
+			tokenSecretName = secret.Name
+			break
+		}
+	}
```

Comment:
like we discussed a last week in slack, we could also create our dedicated named token for the SA [1]
so the we could just get this specific token instead of searching for any of the auto created tokens for
the SA. Do the auto generated ones change?

[1] https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#to-create-additional-api-tokens

Reply:
I looked into this initially and discovered that it just creates a duplicate of the `osp-director-operator-kubevirtagent-token-XXXXX` secret, so that why I just find those secrets and get the token from there.  They don't appear to change unless you uninstall/reinstall the CSV.  But you are right, it is simpler perhaps to just create the duplicate with a known name, instead of getting the SA and then finding the right secret.  I'll rework this piece.